io/include/pcl/io/real_sense_2_grabber.h

```diff
@@ -0,0 +1,213 @@
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#ifndef PCL_IO_REAL_SENSE_2_GRABBER_H
+#define PCL_IO_REAL_SENSE_2_GRABBER_H
+
+#include <pcl/io/boost.h>
+#include <pcl/io/grabber.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+
+
+#include <librealsense2/rs.hpp>
+
+namespace pcl
+{
+
+  /** \brief Grabber for Intel Realsense 2 SDK devices (D400 series)
+    * \note Device width/height defaults to 424/240, the lowest resolutions for D400 devices.
+    * \note Testing on the in_hand_scanner example we found the lower default resolution allowed the app to perform adequately.
+    * \note Developers should use this resolution as a starting point and gradually increase to get the best results
+    * \author Patrick Abadi <patrickabadi@gmail.com>, Daniel Packard <pack3754@gmail.com>
+    * \ingroup io
+  */
+  class PCL_EXPORTS RealSense2Grabber : public pcl::Grabber
+  {
+  public:
+    /** \brief Constructor
+    * \param[in] file_name_or_serial_number used for either loading bag file or specific device by serial number
+    */
+    RealSense2Grabber ( const std::string& file_name_or_serial_number = "", const bool repeat_playback = true );
+
+    /** \brief virtual Destructor inherited from the Grabber interface. It never throws. */
+    virtual ~RealSense2Grabber () throw ();
+
+    /** \brief Set the device options
+    * \param[in] width resolution
+    * \param[in] height resolution
+    * \param[in] fps target frames per second for the device
+    */
+    inline void
+    setDeviceOptions ( uint32_t width, uint32_t height, uint32_t fps = 30 )
+    {
+      device_width_ = width;
+      device_height_ = height;
+      target_fps_ = fps;
+
+      ReInitialize ();
+    }
+
+    /** \brief Start the data acquisition. */
+    virtual void
+    start ();
+
+    /** \brief Stop the data acquisition. */
+    virtual void
+    stop ();
+
+    /** \brief Check if the data acquisition is still running. */
+    virtual bool
+    isRunning () const;
+
+    /** \brief Obtain the number of frames per second (FPS). */
+    virtual float
+    getFramesPerSecond () const;
+
+    /** \brief defined grabber name*/
+    virtual std::string
+    getName () const { return std::string ( "RealSense2Grabber" ); }
+
+    //define callback signature typedefs
+    typedef void (signal_librealsense_PointXYZ) ( const boost::shared_ptr<const pcl::PointCloud<pcl::PointXYZ>>& );
+    typedef void (signal_librealsense_PointXYZI) ( const boost::shared_ptr<const pcl::PointCloud<pcl::PointXYZI>>& );
+    typedef void (signal_librealsense_PointXYZRGB) ( const boost::shared_ptr<const pcl::PointCloud<pcl::PointXYZRGB>>& );
+    typedef void (signal_librealsense_PointXYZRGBA) ( const boost::shared_ptr<const pcl::PointCloud<pcl::PointXYZRGBA>>& );
```

Comment:
It's better to switch these to use `const pcl::PointCloud<pcl::PointXYZ>>::ConstPtr`. That way you'll facilitate our transition into `std::shared_ptr`.